ai21/clients/common/beta/assistant/plans.py

```diff
@@ -15,20 +19,33 @@ def create(
         self,
         *,
         assistant_id: str,
-        code: str,
+        code: str | callable,
+        schemas: list[dict] | list[Type[BaseModel]] | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> PlanResponse:
         pass
 
     def _create_body(
```

Comment:
This function has a lot of logic that I think should be unittested. Can you please add a unittest covering all of the use cases you are supporting/not supporting?